File: two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hashTable = {}
        for x in range(len(nums)):
            val = nums[x]
            if target - val in hashTable:
                return [nums.index(target - val), x]
            if val not in hashTable:
                hashTable[val] = target - val

File: two-sum.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        map<int, int> hashTable;
        for (int i = 0; i < nums.size(); i++)
        {
            if (hashTable.find(nums.at(i)) == hashTable.end())
                hashTable[target - nums.at(i)] = i;
            else
                return {i, hashTable[nums.at(i)]};
        }
        return {0, 0};
    }
};
